thorlcr/activities/thdiskbaseslave.cpp

```diff
@@ -254,6 +241,17 @@ void CDiskReadSlaveActivityBase::start()
     PARENT::start();
     markStart = true;
     diskProgress = 0;
+    unsigned encryptedKeyLen;
+    void *encryptedKey;
+    helper->getEncryptKey(encryptedKeyLen, encryptedKey);
+    if (0 != encryptedKeyLen)
```

Comment:
If this was being used in anger then we would save the previous value and only recreate if the key had changed.